extension/js/common/api/account-servers/flowcrypt-com-api.ts

```diff
@@ -66,29 +66,20 @@ export class FlowCryptComApi extends Api {
 
   public static messageUpload = async (fcAuth: FcUuidAuth | undefined, encryptedDataBinary: Uint8Array, progressCb: ProgressCb): Promise<BackendRes.FcMsgUpload> => {
     const content = new Attachment({ name: 'cryptup_encrypted_message.asc', type: 'text/plain', data: encryptedDataBinary });
-    return await FlowCryptComApi.request<BackendRes.FcMsgUpload>('message/upload', { content, ...(fcAuth || {}) }, 'FORM', undefined, { upload: progressCb });
+    const rawResponse = await FlowCryptComApi.request<{ short: string }>('message/upload', { content, ...(fcAuth || {}) }, 'FORM', undefined, { upload: progressCb });
+    if (!rawResponse.short) {
+      throw new Error('Unexpectedly missing message upload short id');
+    }
+    // careful - this API request returns `url` as well, but that is URL of the S3 object, not of web portal page
+    // therefore we are constructing URL ourselves to point to web portal
+    return { url: `https://flowcrypt.com/${rawResponse.short}` };
```

Comment:
I'm a bit concerned about forming an URL like this.
Shouldn't we percent-encode RFC 3986 section 2.2 Reserved Characters?
Can any of these be contained in `short`?

Reply:
Thank you for noticing. The `short` is returned by our own backend, and contains strictly 10 alphanumeric characters. So for example https://flowcrypt.com/abcde01234